### PR TITLE
#111-[PART 1] Update the mock 1010 form using the finalized pattern

### PR DESCRIPTION
**Pull request by bellepx0**
## Are you removing, renaming or moving a folder in this PR?
- [X] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [X] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

This PR represents Part 1 of the ticket work, with the synthesis of user research findings still in progress. Once the synthesis is complete, we will update the PR with finalized designs and push Part 2.

1. Created a new folder under `patterns/pattern1/ezr` for the mock 10-10ezr form.
2. Based on the `task-green` pattern template, added the necessary pages for the 10-10ezr form, including contact information and home address form pages under the "veteran information" section.
3. Omitted the military service and income/deductions pages, as they do not use prefill functionality.
4. Implemented prefill functionality for the relevant pages.
5. Created two [mock server responses](https://github.com/department-of-veterans-affairs/vets-website/blob/1b908ccace9e27e4e05ebc5f42216ab6a4b8f76d/src/applications/_mock-form-ae-design-patterns/mocks/endpoints/in-progress-forms/mock-form-ae-design-patterns.js#L1-L78) for prefill test cases:
 - `prefillMaximal` includes prefill data for dependents and insurance providers.
 - `prefill` includes no prefill data.
 
### Steps to view 10-10ezr form

1. Pull down branch. In one terminal, run mock server `yarn mock-api --responses src/applications/_mock-form-ae-design-patterns/mocks/server.js`
2. Open another terminal, run `yarn watch --env entry=mock-form-ae-design-patterns`
3. Go to "http://localhost:3001/mock-form-ae-design-patterns/1/ezr". Click on "Start your health benefits update form".
4. Go through form. You should see the prefilled data on the appropriate pages.

## Related issue(s)
https://github.com/department-of-veterans-affairs/tmf-auth-exp-design-patterns/issues/111

## Testing done
Tested locally. 

## Screenshots
Screenshot 2024-09-17 at 11 34 27 AM
Screenshot 2024-09-17 at 11 34 09 AM
Screenshot 2024-09-17 at 11 33 56 AM
Screenshot 2024-09-17 at 11 34 03 AM

## What areas of the site does it impact?
N/A. this is an experimental app that does not touch other areas of the website.

## Acceptance criteria
- [ ] User can access 10-10ezr form through mock form patterns app. 
- [ ] User can confirm that prefilled data is being displayed.

## Quality Assurance & Testing
- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [X] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [X] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by powellkerry — DISMISSED** [review]

**Review by msbtterswrth — COMMENTED** [review]
LGTM! Great work as always!

**Comment by bellepx0** [comment]
> I'm thinking that we should move the components/ezr into one of two places
> 
> * directly into the patterns/pattern1/ezr as a components folder
> * into the shared/components folder
> 
> The reason I think this could be a good idea is that right now the root components folder actually just has one component in it called GetFormHelp, and that could also be moved into either the shared folder or the responsible pattern's folder. That way we dont have a root component folder that conflicts with the shared/components folder and the patterns folder as far as determining where to put things. Thoughts?

Good idea, not sure why I didn't do that initially 🙃. I moved all the files from the root component folder into the  patterns/pattern1/ezr/components folder.

File: src/applications/_mock-form-ae-design-patterns/mocks/endpoints/user/index.js [hunk 1]
```diff
@@ -322,6 +322,20 @@ const baseUserResponses = {
...
+        spouseDateOfBirth: '1990-08-03',
+        spouseSocialSecurityNumber: '451906574',
+        'view:reportDependents': true,
+        'view:skipDependentInfo': false,
+        dependents: [
+          {
+            fullName: { first: 'Mia', last: 'Smith' },
+            dependentRelation: 'Daughter',
+          },
+        ],
```
Comment (adamwhitlock1) [inline src/applications/_mock-form-ae-design-patterns/mocks/endpoints/user/index.js:338]: Does this need to be in the user response? I thought this was something that we put in the in-progress-forms response?

File: src/applications/_mock-form-ae-design-patterns/patterns/pattern1/TaskGreen/config/form.js [hunk 2]
```diff
@@ -68,16 +69,17 @@ const formConfig = {
...
+  // prefillTransformer(pages, formData, metadata) {
+  //   const transformedData = {
+  //     veteranSocialSecurityNumber: formData?.data?.attributes?.veteran?.ssn,
+  //   };
+  //   return {
+  //     metadata,
+  //     formData: transformedData,
+  //     pages,
+  //   };
+  // },
```
Comment (adamwhitlock1) [inline src/applications/_mock-form-ae-design-patterns/patterns/pattern1/TaskGreen/config/form.js:82]: now that we know the prefill transformer can be replaced, I think it is safe to say we can remove this commented out section.

**Review by adamwhitlock1 — DISMISSED** [review]
A couple minor non-blocking things that could be changed, but otherwise lgtm

File: src/applications/_mock-form-ae-design-patterns/patterns/pattern1/TaskGreen/config/form.js (hunk 2 above)
Comment (bellepx0) [reply src/applications/_mock-form-ae-design-patterns/patterns/pattern1/TaskGreen/config/form.js]: deleted this commented out code and the one in TaskYellow config as well.

File: src/applications/_mock-form-ae-design-patterns/mocks/endpoints/user/index.js (hunk 1 above)
Comment (bellepx0) [reply src/applications/_mock-form-ae-design-patterns/mocks/endpoints/user/index.js]: deleted

**Review by adamwhitlock1 — APPROVED** [review]

**Pull request merged**